internal/pkg/gateway/registry.go

```diff
@@ -93,29 +93,51 @@ func (reg *registry) endorsers(channel string, chaincode string) ([]*endorser, e
 				}
 				endpoint := msg.GetAliveMsg().Membership.Endpoint
 
-				groupPeers = append(groupPeers, &endorserState{peer: peer, endpoint: endpoint, height: height})
+				// find the endorser in the registry for this endpoint
+				var endorser *endorser
+				if endpoint == reg.localEndorser.address {
+					endorser = reg.localEndorser
+				} else if e, ok := reg.remoteEndorsers[endpoint]; ok {
+					endorser = e
+				} else {
+					reg.logger.Warnf("Failed to find endorser at %s", endpoint)
+					continue
+				}
+
+				// add to list of endorsing peers, unless endorsingOrgs is specified and it's not in one of the allowed orgs
+				if len(endorsingOrgs) == 0 || contains(endorsingOrgs, endorser.mspid) {
+					groupPeers = append(groupPeers, &endorserState{peer: peer, endorser: endorser, endpoint: endpoint, height: height})
+				}
 			}
+
+			// If the number of available endorsers less than the quantity required, try the next layout
+			if len(groupPeers) < int(quantity) {
+				abandonLayout = true
+				break
+			}
+
 			// sort by decreasing height
 			sort.Slice(groupPeers, sorter(groupPeers, reg.localEndorser.address))
 
 			peerGroup := groupPeers[0:quantity]
 			r = append(r, peerGroup...)
```

Comment:
Not related to this PR but a minor suggestion - A single letter variable works fine if the life of the variable spans a few lines (e.g., `peerGroup`). The variable `r` is declared far above from where it is used so a meaningful name would be better.
```suggestion
			r = append(r, groupPeers[0:quantity]...)
```